tengine: Add modules to passthru

talyz opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change
Follow the interface of the `nginx` package and expose the enabled modules. This is used in the `nginx` module to enable the
`additionalModules` option.

The `nginx-variants`-test broke with the addition of the `additionalModules` option in #116074. This fixes that.


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

talyz committed:
tengine: Add modules to passthru

Follow the interface of the `nginx` package and expose the enabled
modules. This is used in the `nginx` module to enable the
`additionalModules` option.

ajs124 commented:
Do modules actually work with tengine? I guess they should, but maybe somebody should check?

Anyways, with this we should also be able to revert 2d3200e010cc4c6fae62d9f6c31357cb97d606d4.

Thanks for your fast response on this.

talyz committed:
Revert "nixos/nginx: fix eval for tengine"

This reverts commit 2d3200e010cc4c6fae62d9f6c31357cb97d606d4.

talyz committed:
nginx: Add all the nginx tests to passthru.tests

talyz commented:
Yeah, it seems to work fine when I add the `brotli` module to `additionalModules` in `nginx-variants` - all the variants build fine and pass the test. 2d3200e010cc4c6fae62d9f6c31357cb97d606d4 is now reverted. I also added all the nginx tests to `nginx`s `passthru.tests` to make it easier to run all tests in one go.

talyz committed:
tengine: Add passthru.tests
